src/findfile.c

```diff
@@ -1697,27 +1885,37 @@ find_file_in_path_option(
 	{
 	    int		l;
 	    int		run;
+	    size_t	rel_fnamelen = 0;
+	    char_u	*suffix;
 
 	    if (path_with_url(*file_to_find))
 	    {
-		file_name = vim_strsave(*file_to_find);
+		file_name = vim_strnsave(*file_to_find, file_to_findlen);
 		goto theend;
 	    }
 
 	    // When FNAME_REL flag given first use the directory of the file.
 	    // Otherwise or when this fails use the current directory.
+	    if (rel_fname != NULL)
+		rel_fnamelen = STRLEN(rel_fname);
```

Comment:
```suggestion
	    if (rel_fname != NULL)
		rel_fnamelen = STRLEN(rel_fname);
	    // When FNAME_REL flag given first use the directory of the file.
	    // Otherwise or when this fails use the current directory.
```